src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -8115,11 +8148,76 @@ export class StudyViewPageStore
         } else return '';
     }
 
+    readonly survivalEntryMonths = remoteData<
+        { [uniquePatientKey: string]: number } | undefined
+    >({
+        invoke: async () => {
+            const studyIds = this.studyIds;
+            // Please note:
+            // The left truncation adjustment is only available for one study: heme_onc_nsclc_genie_bpc at this time
+            // clinical attributeId still need to be decided in the future
+            if (
+                studyIds.length === 1 &&
+                studyIds[0] === 'heme_onc_nsclc_genie_bpc' &&
+                this.enableLeftTruncation
+            ) {
+                const data = await client.getAllClinicalDataInStudyUsingGET({
+                    attributeId: 'TT_CPT_REPORT_MOS',
+                    clinicalDataType: 'PATIENT',
+                    studyId: studyIds[0],
+                });
+                return data.reduce(
+                    (map: { [patientKey: string]: number }, next) => {
+                        map[next.uniquePatientKey] = parseFloat(next.value);
+                        return map;
+                    },
+                    {}
+                );
+            } else {
+                return undefined;
+            }
+        },
+    });
+
+    readonly isLeftTruncationAvailable = remoteData<boolean>({
+        await: () => [this.survivalEntryMonths],
+        invoke: async () => {
+            return !_.isEmpty(this.survivalEntryMonths.result);
+        },
+    });
+
+    readonly survivalPlotDataWithoutLeftTruncation = remoteData<SurvivalType[]>(
+        {
+            await: () => [
+                this.survivalData,
+                this.selectedPatientKeys,
+                this.survivalPlots,
+                this.survivalEntryMonths,
+            ],
+            invoke: async () => {
+                return this.survivalPlots.result.map(obj => {
+                    obj.survivalData = getPatientSurvivals(
+                        this.survivalData.result,
+                        this.selectedPatientKeys.result!,
+                        obj.associatedAttrs[0],
+                        obj.associatedAttrs[1],
+                        obj.filter,
+                        undefined
+                    );
+                    return obj;
+                });
+            },
+            onError: () => {},
+            default: [],
+        }
+    );
+
     readonly survivalPlotData = remoteData<SurvivalType[]>({
         await: () => [
             this.survivalData,
             this.selectedPatientKeys,
             this.survivalPlots,
+            this.survivalEntryMonths,
         ],
         invoke: async () => {
             return this.survivalPlots.result.map(obj => {
```

Comment:
i can see that this isn't new but it's a little wrong that we are mutating survivalPlots here, tacking on the survival data.   it's suprising to me that typescript allows this.   in other words, that the survivalPlot object will accept  survivalDataWithoutLeftTruncation as a property.   better to just make consumers look it up using plot id?

Reply:
I fixed it by not changing the property in place, is this looks good? https://github.com/cBioPortal/cbioportal-frontend/pull/4324/commits/313fbab0a939a6f75e4e0897e61a1098e2aa546a

Reply:
i don't think this is great either.    it doesn't really make sense to collapse these two types in this way.    maybe just key the survivalPlotData by id and then look it up when we need it?

Reply:
changed to a search by id approach in commit : https://github.com/cBioPortal/cbioportal-frontend/pull/4324/commits/dbdd28d6091e14b3ed8c0712192aef608f512737